### PR TITLE
arch/risc-v: Store/Restore FPU register in exception_common

### PR DESCRIPTION
**Pull request by no1wudi**
## Summary
Move FPU context relative opeartions to exception common
## Impact
RISCV with FPU
## Testing
QEMU

### CODE REVIEW

File: arch/risc-v/src/common/riscv_exception_common.S [hunk 1]
```diff
@@ -147,6 +155,10 @@ exception_common:
   REGLOAD    s0, REG_INT_CTX(sp) /* restore mstatus   */
   csrw       mstatus, s0
 
+#ifdef CONFIG_ARCH_FPU
```
Comment (xiaoxiang781216) [inline arch/risc-v/src/common/riscv_exception_common.S:158]: should we restore before mstatus?

Comment (no1wudi) [reply arch/risc-v/src/common/riscv_exception_common.S:158]: OK, FPU restore would not touch the mstatus, but it's closer to current mainline (restore FPU context first)

Comment (pussuw) [reply arch/risc-v/src/common/riscv_exception_common.S]: Ordering should be:
1. riscv_restorefpu() <- this modifies the context, not the status register first
2. restore mstatus <- restore status register with the updated context

Comment (no1wudi) [reply arch/risc-v/src/common/riscv_exception_common.S]: Yes, the latest change restore the fpu context first and then restore mstatus.

File: arch/risc-v/src/common/riscv_exception_common.S [hunk 2]
```diff
@@ -96,6 +98,11 @@ exception_common:
   addi       s0, sp, XCPTCONTEXT_SIZE
   REGSTORE   s0, REG_X2(sp)       /* original SP      */
 
+#ifdef CONFIG_ARCH_FPU
+  mv         a0, sp
+  jal        x1, riscv_savefpu
+#endif
```
Comment (xiaoxiang781216) [inline arch/risc-v/src/common/riscv_exception_common.S:104]: let's move after line 111 to mirror the restore order.

Comment (no1wudi) [reply arch/risc-v/src/common/riscv_exception_common.S:104]: It's better to keep current location since a0 in line 111 was occupied by mcause.

Comment (xiaoxiang781216) [reply arch/risc-v/src/common/riscv_exception_common.S:104]: But why not move mcause just before line 112?

Comment (no1wudi) [reply arch/risc-v/src/common/riscv_exception_common.S:107]: Done

**Review by xiaoxiang781216 — APPROVED** [review]
LGTM

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**